packages/@react-aria/table/src/useTable.ts

```diff
@@ -70,6 +70,10 @@ export function useTable<T>(props: AriaTableProps<T>, state: TableState<T>, ref:
     gridProps['aria-rowcount'] = state.collection.size + state.collection.headerRows.length;
   }
 
+  if ('expandedKeys' in state) {
+    gridProps.role = 'treegrid';
+  }
```

Comment:
Opted to put tree grid specific aria attributes in table hooks level, not grid hook level. If it turns out that we will have other components that only call the grid hooks and need tree grid aria attributes, then we can refactor.